File: 22.10_PS/1018_카드_정렬하기.py
# boj. 1715

import sys
from heapq import *

input = sys.stdin.readline

answer = 0;
n = int(input())
priority_queue = []
for _ in range(n):
    heappush(priority_queue, int(input()))

while len(priority_queue) > 1:
    t = heappop(priority_queue) + heappop(priority_queue)
    heappush(priority_queue, t)
    answer += t

print(answer)

File: 22.10_PS/1018_카드_정렬하기.java
// boj. 1715

import java.io.*;
import java.util.*;

public class Main {
    static PriorityQueue<Integer> priorityQueue  = new PriorityQueue<>();
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int answer = 0;
        int n = Integer.parseInt(br.readLine());
        priorityQueue = new PriorityQueue<>();

        for (int i = 0; i < n; i++) {
            priorityQueue.add(Integer.parseInt(br.readLine()));
        }

        while (priorityQueue.size() > 1) {
            int target = priorityQueue.poll() + priorityQueue.poll();
            priorityQueue.add(target);
            answer += target;
        }

        System.out.println(answer);
    }
}